microcosm_pubsub/producer.py

```diff
@@ -100,16 +110,32 @@ def publish_message(self, pubsub_message: PubsubMessage):
             **pubsub_message.opaque_data
         )
         self.logger.debug("Publishing message with media type {media_type}", extra=extra)
+        publish_result = "SUCCESS"
+        publish_exception = None
 
         with elapsed_time(extra):
-            result = self.sns_client.publish(
-                TopicArn=pubsub_message.topic_arn,
-                Message=pubsub_message.message,
-                MessageAttributes=pubsub_message.message_attributes,
-            )
+            try:
+                result = self.sns_client.publish(
+                    TopicArn=pubsub_message.topic_arn,
+                    Message=pubsub_message.message,
+                    MessageAttributes=pubsub_message.message_attributes,
+                )
+            except Exception as e:
```

Comment:
Tracking the botocore exception here (after the timeout is done).
We catch it here but throw it later after we send the stats